interpreter.js

```diff
@@ -3226,8 +3226,13 @@ Interpreter.prototype['stepCallExpression'] = function(stack, state, node) {
       state.value = func.nativeFunc.apply(state.funcThis_, state.arguments_);
     } else if (func.asyncFunc) {
       var thisInterpreter = this;
-      var callback = function(value) {
-        state.value = value;
+      var callback = function(value, exception) {
+        if (arguments.length === 1) {
```

Comment:
Hello.

I think this condition makes (maybe unintended) breaking change for **void** async functions.
For example:

```
const { Interpreter } = require('./interpreter');
global.acorn = require('./acorn');

const myCode = `
// something to do
sleep(1000);
// something to do
`;

const initFunc = (i, s) => {
  i.setProperty(s, 'sleep', i.createAsyncFunction((timeMs, callback) => setTimeout(callback, timeMs)));
};
const myInterpreter = new Interpreter(myCode, initFunc);
myInterpreter.run();
```

Above code works well on current JS-Interpreter, but it won't work after apply this PR.

Of course changing `setTimeout(callback, timeMs)` to `setTimeout(() => callback(undefined), timeMs)` works, but presumably it's not good.

Reply:
Strictly speaking there's no such thing as a void function in JavaScript (every function has a return value, even if you don't specify one intentionally, but I take your point about usage compatibility, which is an objective for this PR.  Fortunately it should be easy to fix.

Reply:
Fixed (I believe!)

Thanks for your feedback.